test/fail_compilation/diag8318.d

```diff
@@ -1,7 +1,12 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/diag8318.d(13): Error: function `diag8318.Bar8318.foo` return type inference is not supported if may override base class function
+fail_compilation/diag8318.d(18): Error: function `diag8318.Bar8318.foo` return type inference is not supported if may override base class function
+fail_compilation/diag8318.d(23): Error: function `diag8318.C10021.makeI` return type inference is not supported if may override base class function
+fail_compilation/diag8318.d(31): Error: function `diag8318.Bar10195.baz` return type inference is not supported if may override base class function
+fail_compilation/diag8318.d(37): Error: function `diag8318.B14173.foo` does not override any function
+fail_compilation/diag8318.d(23): Error: class `diag8318.C10021` interface function `I10021 makeI()` is not implemented
+fail_compilation/diag8318.d(29): Error: class `diag8318.Bar10195` interface function `int baz()` is not implemented
```

Comment:
This change was necessary because the order of these error messages changed, when the code which checks if an interface is implemented or not was moved to the second phase of semantic.